### PR TITLE
[MIG] [12.0] purchase_order_type migrate

### PR DESCRIPTION
**Pull request by i-vyshnevska**
migrate https://github.com/OCA/purchase-workflow/tree/11.0/purchase_order_type

### CODE REVIEW

File: purchase_order_type/models/purchase_order.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
-# Copyright (C) 2015 Camptocamp SA
+# Copyright 2015 Camptocamp SA
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
 from odoo import api, fields, models
 from odoo.addons.purchase.models.purchase import PurchaseOrder
 
 
-class PurchaseOrder(models.Model):
+class PurchaseOrderClass(models.Model):
```
Comment (jcoux) [inline purchase_order_type/models/purchase_order.py]:
```suggestion
class PurchaseOrder(models.Model):
```

File: purchase_order_type/models/purchase_order.py [hunk 2]
```diff
@@ -1,11 +1,11 @@
-# Copyright (C) 2015 Camptocamp SA
+# Copyright 2015 Camptocamp SA
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
 from odoo import api, fields, models
 from odoo.addons.purchase.models.purchase import PurchaseOrder
```
Comment (jcoux) [inline purchase_order_type/models/purchase_order.py]:
```suggestion
from odoo.addons.purchase.models.purchase import PurchaseOrder as Purchase
```

File: purchase_order_type/models/purchase_order.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+class PurchaseOrderClass(models.Model):
+    _inherit = 'purchase.order'
+
+    def _default_order_type(self):
+        return self.env['purchase.order.type'].search([], limit=1)
+
+    order_type = fields.Many2one(comodel_name='purchase.order.type',
+                                 readonly=False,
+                                 states=PurchaseOrder.READONLY_STATES,
```
Comment (jcoux) [inline purchase_order_type/models/purchase_order.py]:
```suggestion
                                 states=Purchase.READONLY_STATES,
```

**Review by jcoux — CHANGES_REQUESTED** [review]

**Review by tschanzt — APPROVED** [review]

**Review by jcoux — APPROVED** [review]

**Review by marylla — APPROVED** [review]
Hi there,
I am not sure why the runbot build failed, I could make functional review anyway.
And it worked as expected, correct purchase number sequence was used and the payment term and incoterm was also set according to the type.
LGTM. :)

**Comment by HviorForgeFlow** [comment]
I opened a PR for fix runbot:
https://github.com/OCA/purchase-workflow/pull/849
Once merged you only need to rebase your branch, and we can merge asap.

**Pull request merged**